core/neomutt.c

```diff
@@ -209,3 +210,22 @@ size_t neomutt_mailboxlist_get_all(struct MailboxList *head, struct NeoMutt *n,
 
   return count;
 }
+
+/**
+ * mutt_file_fopen_masked_full - A wrapper around mutt_file_fopen_full with the
+ * mask parameter set to the user's default mode_t permissions, inferred from
+ * the umask
+ */
+FILE *mutt_file_fopen_masked_full(const char *path, const char *mode, const char *file,
+                                  int line, const char *func)
+{
+  // We need to change the current umask to the original one we retrieved from
```

Comment:
**Do** we?
If we're applying the umask to the perms, below, then the file will have the right permissions.

If we don't, we can drop both `umask()` calls.

Reply:
> **Do** we?

I was a bit surprised by this too :) Basically, the `mode_t` that `open` gets, is unmasked unconditionally by the current umask. This is also observed in this [QA](https://stackoverflow.com/questions/9380671/how-to-neglect-the-umask-so-as-to-create-the-file-with-given-permission).

You are correct though, that we don't have to implement the `0666 & ~NeoMutt->user_default_umask` binary operation ourselves, as `open` does it by itself as well, since we set `umask` before calling `open`.